planning/behavior_path_planner/src/behavior_path_planner_node.cpp

```diff
@@ -372,7 +372,7 @@ void BehaviorPathPlannerNode::run()
   const bool is_first_time = !(planner_data_->route_handler->isHandlerReady());
```

Comment:
I suggest using an early return instead of nesting conditions. ie. 

```
if(!route_ptr){
const auto controlled_by_autoware_autonomously =
      planner_data_->operation_mode->mode == OperationModeState::AUTONOMOUS &&
      planner_data_->operation_mode->is_autoware_control_enabled;
    if (!controlled_by_autoware_autonomously && !planner_manager_->hasApprovedModules())
      planner_manager_->resetCurrentRouteLanelet(planner_data_);
return;
}
    planner_data_->route_handler->setRoute(*route_ptr);
    planner_manager_->resetCurrentRouteLanelet(planner_data_);

    // uuid is not changed when rerouting with modified goal,
    // in this case do not need to rest modules.
    const bool has_same_route_id =
      planner_data_->prev_route_id && route_ptr->uuid == planner_data_->prev_route_id;
    // Reset behavior tree when new route is received,
    // so that the each modules do not have to care about the "route jump".
    if (!is_first_time && !has_same_route_id) {
      RCLCPP_INFO(get_logger(), "New uuid route is received. Resetting modules.");
      planner_manager_->reset();
      planner_data_->prev_modified_goal.reset();
    }
```

Reply:
Thank you. You are right we want to avoid nested conditions (https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f56-avoid-unnecessary-condition-nesting) but here we cannot do an early return since we want to execute the rest of the `run()` function in both condition branches.

Reply:
Ah sorry, I got confused, I though it was at the end of the function.